flow360/component/simulation/translator/utils.py

```diff
@@ -0,0 +1,339 @@
+    if isinstance(input_params, SimulationParams):
+        param = input_params
+    elif isinstance(input_params, str):
+        try:
+            # If input is a JSON string
+            param_dict = json.loads(input_params)
+        except json.JSONDecodeError:
+            # If input is a file path
+            with open(input_params, "r", encoding="utf-8") as file:
+                param_dict = json.load(file)
+        if param_dict is None:
+            raise ValueError(f"Invalid input <{input_params}> for translator. ")
+        param = SimulationParams(**param_dict)
+    elif isinstance(input_params, dict):
+        param = SimulationParams(**input_params)
+
+    if param is not None:
+        return param.preprocess(validated_mesh_unit, exclude=preprocess_exclude)
+    raise ValueError(f"Invalid input <{input_params.__class__.__name__}> for translator. ")
+
+
+def replace_dict_key(input_dict: dict, key_to_replace: str, replacement_key: str):
+    """Replace a key in a dictionary."""
+    if key_to_replace in input_dict:
+        input_dict[replacement_key] = input_dict.pop(key_to_replace)
+
+
+def convert_tuples_to_lists(input_dict):
+    """
+    Recursively convert all tuples to lists in a nested dictionary.
+
+    This function traverses through all the elements in the input dictionary and
+    converts any tuples it encounters into lists. It also handles nested dictionaries
+    and lists by applying the conversion recursively.
+
+    Args:
+        input_dict (dict): The input dictionary that may contain tuples.
+
+    Returns:
+        dict: A new dictionary with all tuples converted to lists.
+
+    Examples:
+        >>> input_dict = {
+        ...     'a': (1, 2, 3),
+        ...     'b': {
+        ...         'c': (4, 5),
+        ...         'd': [6, (7, 8)]
+        ...     }
+        ... }
+        >>> convert_tuples_to_lists(input_dict)
+        {'a': [1, 2, 3], 'b': {'c': [4, 5], 'd': [6, [7, 8]]}}
+    """
+    if isinstance(input_dict, dict):
+        return {k: convert_tuples_to_lists(v) for k, v in input_dict.items()}
+    if isinstance(input_dict, tuple):
+        return list(input_dict)
+    if isinstance(input_dict, list):
+        return [convert_tuples_to_lists(item) for item in input_dict]
+    return input_dict
+
+
+def remove_units_in_dict(input_dict):
+    """Remove units from a dimensioned value."""
+    unit_keys = {"value", "units"}
+    if isinstance(input_dict, dict):
+        new_dict = {}
+        if input_dict.keys() == unit_keys:
+            new_dict = input_dict["value"]
+            return new_dict
+        for key, value in input_dict.items():
+            if isinstance(value, dict) and value.keys() == unit_keys:
+                new_dict[key] = value["value"]
+            else:
+                new_dict[key] = remove_units_in_dict(value)
+        return new_dict
+    if isinstance(input_dict, list):
+        return [remove_units_in_dict(item) for item in input_dict]
+    return input_dict
```

Comment:
the old design (Flow360Params) has a check that every unit that is removed from dict is "flow360" unit. Here we just remove units which is dangerous.

Reply:
@benflexcompute

Reply:
@yifan-flex can you send a PR to make sure this is the case?

Reply:
It is now addressed